core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1204,13 +1171,31 @@ class KafkaController(val config: KafkaConfig, zkClient: KafkaZkClient, time: Ti
   }
 
   private def triggerControllerMove(): Unit = {
-    onControllerResignation()
-    activeControllerId = -1
-    zkClient.deleteController()
+    if (!isActive)
+      warn("Controller has already moved when trying to trigger controller movement")
+    else {
+      try {
+        val expectedControllerEpochZkVersion = controllerContext.epochZkVersion
+        activeControllerId = -1
+        onControllerResignation()
+        zkClient.deleteController(expectedControllerEpochZkVersion)
```

Comment:
nits: can we use `controllerContext.epochZkVersion` instead of using `expectedControllerEpochZkVersion` to be consistent with other usage of `controllerContext.epochZkVersion` in this patch?

Reply:
In `onControllerResignation`, `controllerContext` will be reset. So here we need to keep the value before `onControllerResignation` and reuse it in deletion.

Reply:
This makes sense.